app/scripts/fabric-extensions/rescale-2-resize.js

```diff
@@ -40,20 +40,15 @@ var duringResize = {
     this.line(s);
   },
   path: function (s) {
+    var offsetX = s.pathOffset.x;
+    var offsetY = s.pathOffset.y;
     for (var i = 0; i < s.path.length; i++) {
-      s.path[i][1] *= s.scaleX;
-      s.path[i][2] *= s.scaleY;
-      s.path[i][3] *= s.scaleX;
-      s.path[i][4] *= s.scaleY;
+      s.path[i][1] = (s.path[i][1] - offsetX) * s.scaleX + offsetX;
+      s.path[i][2] = (s.path[i][2] - offsetY) * s.scaleY + offsetY;
+      s.path[i][3] = (s.path[i][3] - offsetX) * s.scaleX + offsetX;
+      s.path[i][4] = (s.path[i][4] - offsetY) * s.scaleY + offsetY;
```

Comment:
Would you need to scale offsetX/offsetY when you add it back in?
`(s.path[i][1] - offsetX) * s.scaleX + offsetX*s.scaleX`

I'm not really familiar with what's going on here, so I might be totally off.